trace/trace_test.go

```diff
@@ -0,0 +1,199 @@
+			sampleTrace := &Trace{}
+			sampleTrace.Step(tt.inputString)
+			if sampleTrace.steps[len(sampleTrace.steps)-1].msg != tt.expectedTrace.steps[len(tt.expectedTrace.steps)-1].msg {
+				t.Errorf("Failed %s:\n Expected %v \n Got %v \n", name, tt.expectedTrace, sampleTrace)
+			}
+		})
+	}
+}
+
+func TestTotalTime(t *testing.T) {
+	tests := map[string]*struct {
+		inputTrace *Trace
+	}{
+		"Test 1: Test with current system time": {
+			inputTrace: &Trace{
+				startTime: time.Now(),
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := tt.inputTrace.TotalTime()
+			if got == 0 {
+				t.Errorf("Failed %s:\n Got %v \n", name, got)
+			}
+		})
+	}
+}
+
+func TestLog(t *testing.T) {
+	tests := map[string]*struct {
+		expectedMessages []string
+		sampleTrace      *Trace
+	}{
+		"Test 1: Check for msgs in log": {
+			expectedMessages: []string{
+				"msg1", "msg2", "msg3",
+			},
+			sampleTrace: &Trace{
+				name: "Test1",
+				steps: []traceStep{
+					{time.Now(), "msg1"},
+					{time.Now(), "msg2"},
+					{time.Now(), "msg3"},
+				},
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			var buf bytes.Buffer
+			klog.SetOutput(&buf)
+			tt.sampleTrace.Log()
+			for _, msg := range tt.expectedMessages {
+				if !strings.Contains(buf.String(), msg) {
+					t.Errorf("Test: %s \nMsg %q not found in log: \n%v\n", name, msg, buf.String())
+				}
+			}
+		})
+	}
+}
+
+func TestLogIfLong(t *testing.T) {
+	currentTime := time.Now()
+	type mutate struct {
+		delay time.Duration
+		msg   string
+	}
+
+	tests := map[string]*struct {
+		expectedMessages []string
+		sampleTrace      *Trace
+		threshold        time.Duration
+		mutateInfo       []mutate // mutateInfo contains the information to mutate step's time to simulate multiple tests without waiting.
+
+	}{
+		"Test 1 :": {
+			expectedMessages: []string{
+				"msg2",
+			},
+			mutateInfo: []mutate{
+				{10, "msg1"},
+				{1000, "msg2"},
+				{0, "msg3"},
+			},
+			threshold: 500,
+		},
+		"Test 2 :": {
+			expectedMessages: []string{
+				"msg3",
+			},
+			mutateInfo: []mutate{
+				{0, "msg1"},
+				{0, "msg2"},
+				{50, "msg3"},
+			},
+			threshold: 10,
+		},
+		"Test 3 :": {
+			expectedMessages: []string{
+				"msg1", "msg2", "msg3",
+			},
+			mutateInfo: []mutate{
+				{0, "msg1"},
+				{0, "msg2"},
+				{0, "msg3"},
+			},
+			threshold: 0,
+		},
+		"Test 4 :": {
+			expectedMessages: []string{},
+			mutateInfo: []mutate{
+				{10, "msg1"},
+				{0, "msg2"},
+				{0, "msg3"},
+			},
+			threshold: 20,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			var buf bytes.Buffer
+			klog.SetOutput(&buf)
+
+			tt.sampleTrace = New("Test trace")
+
+			for index, mod := range tt.mutateInfo {
+				tt.sampleTrace.Step(mod.msg)
+				tt.sampleTrace.steps[index].stepTime = currentTime.Add(mod.delay)
+			}
+
+			tt.sampleTrace.LogIfLong(tt.threshold)
+
+			fmt.Printf("%s\n\n", buf.String())
```

Comment:
Can we remove this? Guessing it was left behind from debugging the test earlier

Reply:
yeah really thanks for pointing out